src/webob/acceptparse.py

```diff
@@ -268,6 +667,1250 @@ def _match(self, mask, item):
+                # asterisk_non0_found
+            else:  # {non-* range};q={not 0}
+                acceptable_ranges.append((range_, qvalue))
+        # Sort acceptable_ranges by qvalue, descending order
+        acceptable_ranges.sort(key=lambda tuple_: tuple_[1], reverse=True)
+        # Sort guaranteed to be stable with Python >= 2.2, so position in
+        # header is tiebreaker when two ranges have the same qvalue
+
+        acceptable_ranges = [tuple_[0] for tuple_ in acceptable_ranges]
+        lowered_tags = [tag.lower() for tag in tags]
+
+        def best_match(range_):
+            subtags = range_.split('-')
+            while True:
+                for index, tag in enumerate(lowered_tags):
+                    if tag in not_acceptable_ranges:
+                        continue
+                        # We think a non-'*' range with q=0 represents only
+                        # itself as a tag, and there should be no falling back
+                        # with subtag truncation. For example, with
+                        # 'en-gb;q=0', it should not mean 'en;q=0': the client
+                        # is unlikely to expect that specifying 'en-gb' as not
+                        # acceptable would mean that 'en' is also not
+                        # acceptable. There is no guidance on this at all in
+                        # the RFCs, so it is left to us to decide how it should
+                        # work.
+
+                    if tag == range_:
+                        return tags[index]  # return the pre-lowered tag
+
+                try:
+                    subtag_before_this = subtags[-2]
+                except IndexError:  # len(subtags) == 1
+                    break
+                # len(subtags) >= 2
+                if len(subtag_before_this) == 1 and (
+                    subtag_before_this.isdigit() or
+                    subtag_before_this.isalpha()
+                ):  # if subtag_before_this is a single-letter or -digit subtag
+                    subtags.pop(-1)  # pop twice instead of once
+                subtags.pop(-1)
+                range_ = '-'.join(subtags)
+
+        for range_ in acceptable_ranges:
+            match = best_match(range_=range_.lower())
+            if match is not None:
+                return match
+
+        if not asterisk_q0_found:
+            if default_range is not None:
+                lowered_default_range = default_range.lower()
+                match = best_match(range_=lowered_default_range)
+                if match is not None:
+                    return match
+
+            if default_tag is not None:
+                lowered_default_tag = default_tag.lower()
+                if lowered_default_tag not in not_acceptable_ranges:
+                    return default_tag
+
+        try:
+            return default()
+        except TypeError:  # default is not a callable
+            return default
+
+    def quality(self, offer, modifier=1):
```

Comment:
> Return the quality of the given offer. Returns None if there is no match (not 0).

That's the docs on `.quality()` at the moment. Feel free to remove `modifier` since it is not documented what it does, or why it is set to 1 by default.

Set it to 1.